src/coord/src/catalog/storage.rs

```diff
@@ -519,27 +591,30 @@ impl<S: Append> Connection<S> {
         Ok(GlobalId::User(id))
     }
 
-    pub async fn allocate_compute_instance_id(&mut self) -> Result<ComputeInstanceId, Error> {
-        let id = self.allocate_id("compute", 1).await?;
-        let id = id.into_element();
-        Ok(id)
-    }
-
     async fn allocate_id(&mut self, id_type: &str, amount: u64) -> Result<Vec<u64>, Error> {
+        let mut tx = self.transaction().await?;
```

Comment:
I have a preference to switch this back to upsert_key to exercise that code path a bit more. I think it's a bit faster too because it avoids iterating over each k,v pair too. (Although there are so few here, and they are all cached anyway so, maybe this isn't a realistic concern.)

Reply:
I agree, I had a commit for this but forgot to push it. We'll just wait to merge this until the other issue is fixed.